compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/symbols/ObjectTypeSymbol.java

```diff
@@ -50,4 +50,12 @@
      * @return The list of included object types
      */
     List<TypeSymbol> typeInclusions();
+
+    /**
+     * Checks whether this object type can be used in a listener declaraion. i.e., whether this object type is a subtype
+     * of the builtin `Listener` type.
+     *
+     * @return Returns true if this type is a subtype of `Listener`
+     */
+    boolean isCompatibleListenerType();
```

Comment:
Another option is we can introduce another utility Class to check this. I guess we have similar validation in other places too.

Reply:
True, but the issue is, where would such a util fit in? This is sort of like the same discussion we had about the check for assignability and lang lib methods.